File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/service/TripsInteractor.java
package com.worldventures.dreamtrips.modules.trips.service;

import com.worldventures.dreamtrips.core.janet.SessionActionPipeCreator;
import com.worldventures.dreamtrips.modules.trips.command.GetTripDetailsCommand;
import com.worldventures.dreamtrips.modules.trips.command.GetTripsCommand;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.techery.janet.ActionPipe;
import rx.schedulers.Schedulers;

@Singleton
public class TripsInteractor {

   private final ActionPipe<GetTripDetailsCommand> detailsPipe;
   private final ActionPipe<GetTripsCommand> tripsPipe;

   @Inject
   public TripsInteractor(SessionActionPipeCreator sessionActionPipeCreator) {
      detailsPipe = sessionActionPipeCreator.createPipe(GetTripDetailsCommand.class, Schedulers.io());
      tripsPipe = sessionActionPipeCreator.createPipe(GetTripsCommand.class, Schedulers.io());
   }

   public ActionPipe<GetTripDetailsCommand> detailsPipe() {
      return detailsPipe;
   }

   public ActionPipe<GetTripsCommand> tripsPipe() {
      return tripsPipe;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/service/TripsInteractor.kt
package com.worldventures.dreamtrips.modules.trips.service

import com.worldventures.core.janet.SessionActionPipeCreator
import com.worldventures.dreamtrips.modules.trips.service.command.CheckTripsByUidCommand
import com.worldventures.dreamtrips.modules.trips.service.command.GetTripDetailsCommand
import com.worldventures.dreamtrips.modules.trips.service.command.GetTripsByUidCommand
import com.worldventures.dreamtrips.modules.trips.service.command.GetTripsCommand
import com.worldventures.dreamtrips.modules.trips.service.command.GetTripsFilterDataCommand
import com.worldventures.dreamtrips.modules.trips.service.command.GetTripsLocationsCommand
import com.worldventures.dreamtrips.modules.trips.service.command.TripFiltersAppliedCommand
import rx.schedulers.Schedulers

class TripsInteractor(pipeCreator: SessionActionPipeCreator) {

   val tripsPipe = pipeCreator.createPipe(GetTripsCommand::class.java, Schedulers.io())
   val detailsPipe = pipeCreator.createPipe(GetTripDetailsCommand::class.java, Schedulers.io())
   val tripsByUidPipe = pipeCreator.createPipe(GetTripsByUidCommand::class.java, Schedulers.io())
   val checkTripsByUidPipe = pipeCreator.createPipe(CheckTripsByUidCommand::class.java, Schedulers.io())
   val mapObjectsActionPipe = pipeCreator.createPipe(GetTripsLocationsCommand::class.java, Schedulers.io())
   val tripFiltersPipe = pipeCreator.createPipe(GetTripsFilterDataCommand::class.java, Schedulers.io())
   val tripFiltersAppliedPipe = pipeCreator.createPipe(TripFiltersAppliedCommand::class.java, Schedulers.io())

   init {
      tripFiltersAppliedPipe.send(TripFiltersAppliedCommand())
   }
}
